### ISSUE TITLE
Addon does not work with XBOX One

### ISSUE THREAD

**Issue by zach-morris** [zach-morris/plugin.program.iagl#42]
The addon does not yet work with XBOX One.  Main reasons:
I dont have an xbox one, so I cant develop for it
Until recently, retroarch and Kodi weren't even available for xbox one.  This has [recently](https://www.reddit.com/r/RetroArch/comments/ajbj5s/retroarch_for_xbox_one_is_available_for_download/) [changed](https://kodi.tv/article/kodi-xbox-one).

If there is someone with an XBOX One interested in helping me develop for the system, please send me a note!

**Comment by zach-morris** [comment]
Well, first you'd need both Kodi and Retroarch installed, then it's really only a matter of validating the required command to send to launch a game.
Some info on how that works is [here](https://blogs.windows.com/buildingapps/2017/07/05/command-line-activation-universal-windows-apps/), and [here](https://stackoverflow.com/questions/44031891/automate-uwp-camera-app)

From IAGL, you would enter a manual command line entry something like:
```
RetroArch /L "path_to_game"
```
In a perfect world it would be that easy.  One I don't know what the command line format is/should be.  Second I don't even know if UWP allows for execution of commands like this at all.  There would be some experimentation required for both.  Maybe someone else out there has suggestions on what the command line format might be?

**Comment by krzys-h** [comment]
1. While both RetroArch and Kodi work on the Xbox One, you can't use RetroArch inside Kodi there yet AFAIK. I doubt this will change due to legal reasons - Microsoft does not allow emulators on the Store, so if Kodi wants to stay on the store it can't have support for any emulation features in the UWP version and RetroArch is only available if you run the console in dev mode and sideload the app manually.
2. UWP is very sandboxed, you can't simply launch a command, you can't even access most of the filesystem using standard Win32 APIs. The only way I can think of would be to launch the other app with an URI handler (`retroarch://something/here/but/probably/not/file/path/because/sandboxing`).

**Comment by krzys-h** [comment]
Oh, and I forgot to mention: if you don't have an Xbox to test, the UWP version on Windows 10 runs 99.9% the same as it would on Xbox (that's why they call it UNIVERSAL Windows Platform :P)

**Comment by drinfernoo** [comment]
@krzys-h Kodi being in the store probably has very little to do with emulation, since Kodi 18.0 now comes bundled with libretro compatibility, and regardless, IAGL is only a conduit through which Kodi can launch RetroArch.

**Comment by zach-morris** [comment]
Thanks @krzys-h, understood dev mode would be a requirement.  I'll look into UWP on windows

**Comment by Doctor1577** [comment]
Am working on it

**Comment by Doctor1577** [comment]
Working on te bundling non store kodi version to appx bundle will let you know

**Comment by weednbeer** [comment]
hi everyone ... i was wondering if in developer mode it was possible to add emulators in the kodi add ons section ... should it do that? I know that Microsoft is blocking on the store version but on a developer version it might be possible ??

**Comment by binoeza** [comment]
First at all, sry my bad English!

I try a method without Dev mode to grab the retroArch! Only download the portable version, put it in a external drive (like an SD or pendrive), configure in IAGL and "voilà"! But can't push the command to execute the ROMs... I'm a completely noob in programing commands, so can't help more than it! If someone have skills to write the commands, and try it! Send me a message

**Comment by lukeraw93** [comment]
Hi all,  (noob alert) please forgive.

After some intresting messing around over the last couple of days i have managed to install KODI with IAGL & Retroarch onto my Xbox in dev mode. i have succesfully downloaded a ROM via IAGL and rather than booting retroarch through kodi, i ran it standalone and was able to use a USB drive to store roms from IAGL downloaded via the xbox to be ran in the retroarch app seperatley.

The only issue im currently encountering and i suspect this is due to the drive being external is the VFS error when trying to play larger games, So far i have ran sega mega drive roms with no problems. i also managed to run a "small" playstation one game. 

In theory i was going to change the download path in KODI to a location on the Xbox where i could then use retroarch to run them seperatley, however due to the nature of the UWP system the only place i can get retroarch to locate that KODI also has access to is an external device. So far i have only used a USB pendrive, i have toyed with the idea of tring it with an external HDD but i fear the same thing would happen with the VFS error.

So all in all, ive a small work around for smaller games but thats all.

P.S forgive me if this is already known and used method, i am very new to this.

Thanks

**Comment by lukeraw93** [comment]
> The addon does not yet work with XBOX One (XBONE). Main reasons:
> I dont have an xbox one, so I cant develop for it
> Until recently, retroarch and Kodi weren't even available for xbox one. This has [recently](https://github.com/libretro/RetroArch/pull/7834/commits/6038060cd7e74df4cb36c50460ce9b9654c1281c) [changed](https://kodi.tv/article/kodi-xbox-one).
> 
> If there is someone with an XBOX One interested in helping me develop for the system, please send me a note!

**Comment by zach-morris** [comment]
@binoeza , yes it's the command that is explicitly the problem.  Kodi cannot execute external commands on an XBOX given the sandbox security in place.

As you and @lukeraw93 mentioned, you should be able to download games to a location and then open RA and launch them from there.  V19 of IAGL should be able to save to network locations (like an SMB network share) as well, so if the XBOX can mount a network share that might be another way to get it to work.

**Comment by lukeraw93** [comment]
@zach-morris yeah not a bad idea, it may eliminte the VFS error and would be quicker than having to upload via the remote connection feature on xbox, i'll try and set up a network location from windows 10 and see if the file explorer XBOX Dev mode defaults to will recognize it, i'll update this as soon as i know the result.

**Comment by lukeraw93** [comment]
> Working on te bundling non store kodi version to appx bundle will let you know
@Doctor1577 
Did you manage to develop this? 

@weednbeer 
I found a list of test-builds including UWP if thats any use to anyone?
http://ftp.igh.cnrs.fr/pub/xbmc/
Maybe one of these has emulator support?

**Comment by zach-morris** [comment]
Within Kodi directly, I've found that there is a VFS addon for UWP for SMB support.  That doesn't help outside of Kodi however.  In theory that would let you connect to the share within Kodi, and if you were using Kodi game addons instead of RA, then it would all work.  I'm not sure if there is any official source for UWP binary game addons for kodi.

I have an unofficial place for kodi game addons, see [here](https://github.com/zach-morris/kodi_libretro_buildbot_game_addons#kodi_libretro_buildbot_game_addons-for-kodi).  However, I'm not sure what the requirements are for a UWP binary addon and if 'windows' is just a drop in replacement for them or not, but I can look into it.
